Student/Student/Submissions/SubmissionRubric/RubricCircleView.swift

```diff
@@ -17,15 +17,198 @@
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        setup()
+
+        if let rubric = rubric {
+            circleViewHeightConstraint?.constant = RubricCircleView.computedHeight(rubric: rubric, maxWidth: bounds.size.width)
+        }
+    }
+
+    func setup() {
+        if circleView == nil {
+            circleView = RubricCircleView(frame: CGRect.zero)
+            circleView?.rubric = rubric
+            circleView?.courseColor = courseColor
+            circleView?.buttonClickDelegate = self
+            addSubview(circleView!)
+            circleView?.pinToLeftAndRightOfSuperview()
+            circleView?.addConstraintsWithVFL("V:|[view]", views: nil, metrics: nil)
+            circleViewHeightConstraint = circleView?.addConstraintsWithVFL("V:[view(50)]", views: nil, metrics: nil)?.first
+        }
+
+        if headerContainer == nil {
+            let margin = RubricCircleViewWithDescription.margin
+            let verticalMarginMetrics = ["margin": margin / 2]
+            let horizontalMarginMetrics = ["margin": margin]
+
+            headerContainer = UIView(frame: CGRect.zero)
+            addSubview(headerContainer!)
+            headerContainer?.pinToLeftAndRightOfSuperview()
+            guard let circleView = circleView else { return }
+            headerContainer?.addConstraintsWithVFL("V:[circles]-(circleViewSpace)-[view]", views: ["circles": circleView], metrics: ["circleViewSpace": RubricCircleView.space])
+            headerContainerHeightConstraint = headerContainer?.addConstraintsWithVFL("V:[view(70)]")?.first
+            headerContainer?.layer.cornerRadius = 8
+            headerContainer?.layer.masksToBounds = true
+
+            guard let container = headerContainer else { return }
+
+            header = DynamicLabel(frame: CGRect.zero)
+            header?.font = UIFont.scaledNamedFont(.semibold14)
+            container.addSubview(header!)
+            header?.addConstraintsWithVFL("H:|-(margin)-[view]-(margin)-|", metrics: horizontalMarginMetrics)
+            header?.addConstraintsWithVFL("V:|-(margin)-[view(21)]", metrics: verticalMarginMetrics)
+
+            subHeader = DynamicLabel(frame: CGRect.zero)
+            subHeader?.numberOfLines = 0
+            subHeader?.font = RubricCircleViewWithDescription.subHeaderFont
+            container.addSubview(subHeader!)
+            subHeader?.addConstraintsWithVFL("H:|-(margin)-[view]-(margin)-|", metrics: horizontalMarginMetrics)
+            subHeader?.addConstraintsWithVFL("V:[header]-(margin)-[view]", views: ["header": header!], metrics: ["margin": 1])
+            subHeaderHeightConstraint = subHeader?.addConstraintsWithVFL("V:[view(21)]")?.first
+
+            container.backgroundColor = courseColor.withAlphaComponent(rubricCircleViewAlphaColor)
+
+            if let rubric = rubric {
+                didClickRating(atIndex: rubric.selectedIndex)
+            }
+        }
+    }
+
+    func updateHeights() {
```

Comment:
Why are we calculating all these heights ourselves? Wouldn't it be easier to use a xib or a UIStackView and let auto layout figure out the heights?